climada/hazard/base.py

```diff
@@ -713,12 +714,20 @@ def from_raster_xarray(
             data[coords["latitude"]].values, data[coords["longitude"]].values, crs=crs,
         )
 
-        def to_csr_matrix(array: np.ndarray) -> sparse.csr_matrix:
+        def to_csr_matrix(array: xr.DataArray) -> sparse.csr_matrix:
             """Store a numpy array as sparse matrix, optimizing storage space
 
             The CSR matrix stores NaNs explicitly, so we set them to zero.
             """
-            return sparse.csr_matrix(np.where(np.isnan(array), 0, array))
+            array = array.where(array.notnull(), 0)
+            array = xr.apply_ufunc(
+                sp.COO.from_numpy,
+                array,
+                dask="parallelized",
+                output_dtypes=[array.dtype]
+            )
+            array = array.compute().data  # Load into memory, now it's a sparse.COO array
+            return array.tocsr()  # Convert sparse.COO to scipy.sparse.csr_matrix
```

Comment:
```suggestion
            sparse_coo = array.compute().data  # Load into memory
            return sparse_coo.tocsr()  # Convert sparse.COO to scipy.sparse.csr_matrix
```
Just because variables changing their type in the middle of a method make me uneasy. Or is there a reason for it?

Reply:
No, not at all 😁